src/Common/src/Interop/Windows/sspicli/NegotiationInfoClass.cs

```diff
@@ -17,14 +17,21 @@ internal class NegotiationInfoClass
 
         internal NegotiationInfoClass(SafeHandle safeHandle, int negotiationState)
         {
+            bool globalLogEnabled = GlobalLog.IsEnabled;
```

Comment:
@stephentoub I've discussed this offline with @josguil. We'd prefer us using the recommended pattern above instead of cashing the IsEnabled() result. 
Unfortunately that means redesign of all EventSource related classes. Please open a tracking issue marked with the [System.Net](https://github.com/dotnet/corefx/labels/System.Net) label.

One scenario where the code as written in this PR would cause issues is when logging is enabled for a short period of time during run-time. If the value is cashed, we could end up having confusing logs where none of the inner functions recorded any events while the outer function logged an enter and exit.